src/ProjectTemplates/test/Templates.Mvc.Tests/RazorPagesTemplateTest.cs

```diff
@@ -309,8 +309,8 @@ private async Task<Project> BuildAndPublishRazorPagesTemplateIdentityWeb(string
         Assert.True(0 == buildResult.ExitCode, ErrorMessages.GetFailedProcessMessage("build", project, buildResult));
 
         // Publish builds in "release" configuration. Running publish should ensure we can compile in release and that we can publish without issues.
-        buildResult = await project.RunDotNetPublishAsync();
-        Assert.True(0 == buildResult.ExitCode, ErrorMessages.GetFailedProcessMessage("publish", project, buildResult));
+        var publishResult = await project.RunDotNetPublishAsync();
```

Comment:
What if we instead made `RunDotNetPublishAsync` and `RunDotNetBuildAsync` do the assert check? Then there isn't a chance to grab the wrong process

Reply:
Yeah I like this better, at least log the error messages inside of the Publish/BuildAsync, even if we do the assert outside